packages/amplify_auth_cognito/ios/Classes/Utils/AuthCodeDeliveryDetailsSerialization.swift

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import Foundation
+import Amplify
+
+func serializeAuthCodeDeliveryDetails(deliveryDetails: AuthCodeDeliveryDetails) -> [String: String] {
```

Comment:
Note: Same note I made on the Android side - Right now this is only used for update user attributes (single update and batch update) but this should be reusable for sign up, sign in, reset password, and resend sign up code. I was hesitant to refactor all of those in this PR though.

Reply:
Agreed - this should be a seperate PR.  There will be some conflicts with NS, so maybe we can make these changes at that time.